### PR TITLE
Initial edit account view

### PR DESCRIPTION
**Pull request by nikolalukovic**

### CODE REVIEW

File: app/src/main/kotlin/net/primal/android/navigation/PrimalAppNavigation.kt [hunk 1]
```diff
@@ -274,6 +278,8 @@ fun PrimalAppNavigation() {
                 navController = navController,
             )
 
+            editProfile(route = "edit-profile", navController = navController)
```
Comment (AleksandarIlic) [inline app/src/main/kotlin/net/primal/android/navigation/PrimalAppNavigation.kt:281]: edit_profile

File: app/src/main/kotlin/net/primal/android/profile/details/ProfileContract.kt [hunk 2]
```diff
@@ -11,6 +11,7 @@ interface ProfileContract {
     data class UiState(
         val profileId: String,
         val isProfileFollowed: Boolean,
+        val isCurrentUser: Boolean,
```
Comment (AleksandarIlic) [inline app/src/main/kotlin/net/primal/android/profile/details/ProfileContract.kt:14]: isActiveUser

File: app/src/main/kotlin/net/primal/android/profile/details/ProfileScreen.kt [hunk 3]
```diff
@@ -601,9 +613,17 @@ private fun ProfileActions(
...
+            when (isFollowed) {
+                true -> UnfollowButton(onClick = onUnfollow)
+                false -> FollowButton(onClick = onFollow)
+            }
+        }
+        Spacer(modifier = Modifier.width(8.dp))
+        if (isCurrentUser) {
+            EditProfileButton(onClick = {
+                onEditProfileClick()
+            })
```
Comment (AleksandarIlic) [inline app/src/main/kotlin/net/primal/android/profile/details/ProfileScreen.kt:626]: You either need to show Follow/Unfollow buton or Edit Profile. If/else should be enough.

File: app/src/main/kotlin/net/primal/android/profile/edit/EditProfileScreen.kt [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    state: EditProfileContract.UiState,
+    eventPublisher: (EditProfileContract.UiEvent) -> Unit,
+    paddingValues: PaddingValues,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .systemBarsPadding()
+            .navigationBarsPadding()
```
Comment (AleksandarIlic) [inline app/src/main/kotlin/net/primal/android/profile/edit/EditProfileScreen.kt:96]: Remove systemBarsPadding() and navigationsBarsPadding(). It's causing padding issues.

File: app/src/main/kotlin/net/primal/android/profile/edit/EditProfileContract.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        data class AboutMeChangedEvent(val aboutMe: String) : UiEvent()
+        data class LightningAddressChangedEvent(val lightningAddress: String) : UiEvent()
+        data class Nip05IdentifierChangedEvent(val nip05Identifier: String) : UiEvent()
+        data class AvatarUriChangedEvent(val avatarUri: Uri?) : UiEvent()
+        data class BannerUriChangedEvent(val bannerUri: Uri?) : UiEvent()
+        data object SaveProfileEvent : UiEvent()
+    }
+}
+
+fun EditProfileContract.UiState.toProfileMetadata(): ProfileMetadata {
```
Comment (AleksandarIlic) [inline app/src/main/kotlin/net/primal/android/profile/edit/EditProfileContract.kt:39]: This shouldn't be in Contract. Move it to viewModel.

File: app/src/main/kotlin/net/primal/android/profile/edit/EditProfileContract.kt [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    return ProfileMetadata(
+        displayName = this.displayName,
+        handle = this.name,
+        website = this.website,
+        about = this.aboutMe,
+        lightningAddress = this.lightningAddress,
+        nostrVerification = this.nip05Identifier,
+        picture = this.avatarUri,
+        banner = this.bannerUri
+    )
```
Comment (AleksandarIlic) [inline app/src/main/kotlin/net/primal/android/profile/edit/EditProfileContract.kt:49]: We need to make sure that we don't send empty strings. You can use `.ifEmpty { null }` extension function to return null if the fields are not nullables.

File: app/src/main/kotlin/net/primal/android/profile/edit/EditProfileScreen.kt [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+    eventPublisher: (EditProfileContract.UiEvent) -> Unit,
+    onClose: () -> Unit
+) {
+    val snackbarHostState = remember { SnackbarHostState() }
+    ErrorHandler(error = state.error, snackbarHostState = snackbarHostState)
+
+    Scaffold(
+        topBar = {
+            PrimalTopAppBar(
+                title = "Edit Profile",
```
Comment (AleksandarIlic) [inline app/src/main/kotlin/net/primal/android/profile/edit/EditProfileScreen.kt:68]: Move to strings.xml.

File: app/src/main/kotlin/net/primal/android/profile/edit/EditProfileViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+            setState {
+                copy(
+                    displayName = it.metadata?.displayName ?: "",
+                    name = it.metadata?.handle ?: "",
+                    aboutMe = it.metadata?.about ?: "",
+                    website = it.metadata?.website ?: "",
+                    lightningAddress = it.metadata?.lightningAddress ?: "",
+                    nip05Identifier = it.metadata?.internetIdentifier ?: "",
+                    bannerUri = it.metadata?.banner?.toUri(),
+                    avatarUri = it.metadata?.picture?.toUri()
```
Comment (AleksandarIlic) [inline app/src/main/kotlin/net/primal/android/profile/edit/EditProfileViewModel.kt:89]: Regarding the situation where we can have both local and remote images.

We could probably update `ProfileMetadata` domain class to have `localPictureUri`, `localBannerUri`, `remotePictureUrl` and `remoteBannerUrl`, and then handle all four potential use cases where appropriate. 

You will also need to update the `ProfileRepository` to fallback to `remote` urls if there are no local one provided.

**Review by AleksandarIlic — CHANGES_REQUESTED** [review]
Overall looks very good. Just few minor things.

File: app/src/main/kotlin/net/primal/android/profile/edit/EditProfileContract.kt (hunk 6 above)
Comment (nikolalukovic) [reply app/src/main/kotlin/net/primal/android/profile/edit/EditProfileContract.kt:49]: It's a `Uri`. I don't think `Uri` could be created with an empty string. Both are already nullable.

Comment (AleksandarIlic) [reply app/src/main/kotlin/net/primal/android/profile/edit/EditProfileContract.kt]: I was referring to String values. For example if there was no nip05 value in metadata and no nip05 value in editor, we would submit nostr event with nip05="".

**Pull request merged**